packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -299,6 +299,7 @@ class AOTSnapshotter {
       TargetPlatform.ios,
       TargetPlatform.darwin_x64,
       TargetPlatform.linux_x64,
+      TargetPlatform.linux_arm64,
```

Comment:
We're trying to move away from the giant switch of platforms. The problem this causes is that it makes a lot of invalid states easily representable.


Instead this should be factored out into a LinuxArch enum. This could be done before any of the work to support arm

Reply:
@jonahwilliams 
Okay. I will create a LinuxArch enum like Android/iOS and modify this PR.

Reply:
Looking at the cache APIs we have today, it doesn't seem easy to do this. I wonder if you've had any progress? Perhaps we should reconsider for now

Reply:
@jonahwilliams 
I'm reading around the source code of cache.dart. I'm also considering a fix to support ARM64 Host. Please give me some time.

Reply:
@jonahwilliams 
By the way, I think `TargetPlatform` should be separated into OS and Arch. Are you already considering some kind of refactoring? It may take some time, but is it okay for me to consider it?

Reply:
Sorry I missed this - that is probably a good change but its a bit out of scope

Reply:
OK. Thank you.